ws-client/src/transport.rs

```diff
@@ -163,8 +163,8 @@ pub async fn websocket_connection(config: WsConfig<'_>) -> Result<(Sender, Recei
 	for sockaddr in target.to_socket_addrs().await.map_err(WsHandshakeError::ResolutionFailed)? {
 		let builder = WsTransportClientBuilder {
 			target: sockaddr,
-			host: host.into(),
-			dns_name: target.as_str().into(),
+			host: target.as_str().into(),
```

Comment:
I fucked up this,  
`host` - hostname with port kusama-rpc.polkadot.io:443
`dns-name` - kusama-rpc.polkadot.io

before this PR it was the opposite which made it fail on bad DNS name.

Reply:
I think having `host` and `dns_name` is confusing and leads to some funny looking code elsewhere:

```rust
						let dns_name = webpki::DNSNameRef::try_from_ascii_str(&self.dns_name)?;
						let tls_stream = connector.connect(&dns_name.to_owned(), socket?).await?;
```
…after all `DNSName` is just a newtype around `String`?

What I'm trying to say is that as far as I can see we only use `dns_name` in `WsTransportClientBuilder.build` which is not performance critical (and when we do use it, we do that conversion to `DNSNameRef` and then clone back to a `String`). So perhaps we can just store host and port and skip the `dns_name`?

Reply:
> So perhaps we can just store host and port and skip the dns_name?

Sounds good to me, I did least effort to get it working and the bug was introduced by the naming. Let's do it properly ^^